chapters/arrays.tex

```diff
@@ -195,10 +195,14 @@ \section{Built-in Array Functions}\doublelabel{built-in-array-functions}
 \lstinline[mathescape=true]!size(C, $j$)! = $1$ for $n_{\mathrm{A}} + 1 \leq j \leq n$, \lstinline[mathescape=true]!C[$i_{1}$, $\ldots$, $i_{n_{\mathrm{A}}}$, 1, $\ldots$, 1]! =
 \lstinline[mathescape=true]!A[$i_{1}$, $\ldots$, $i_{n_{\mathrm{A}}}$]!\\ \hline
 \end{longtable}
+The argument $n$ must be literal or a constant that can be evaluated during translation because it determines
+the number of dimensions of the returned array.
 
 \begin{nonnormative}
-The function \lstinline!promote! cannot be used in Modelica, because the number of dimensions of the returned array cannot be determined at compile time
-if $n$ is a variable.  Below, \lstinline!promote! is only used for constant $n$.
+Below, \lstinline!promote! is only used for constant $n$.
```

Comment:
This doesn't sound too reassuring.  Yes, we know that n has to be constant, but we also know that it must be possible to evaluate during translation, and this is the kind of guarantee we need here.